bin/verdicts.py

```diff
@@ -0,0 +1,452 @@
+from pathlib import Path
+import shutil
+import sys
+from enum import Enum
+
+from util import ProgressBar
+import config
+from colorama import Fore, Style
+
+
+class Verdict(Enum):
+    """The verdict of a testcase or testgroup"""
+
+    ACCEPTED = 1
+    WRONG_ANSWER = 2
+    TIME_LIMIT_EXCEEDED = 3
+    RUNTIME_ERROR = 4
+    VALIDATOR_CRASH = 5
+    COMPILER_ERROR = 5
+
+    def __str__(self):
+        return {
+            Verdict.ACCEPTED: 'ACCEPTED',
+            Verdict.WRONG_ANSWER: 'WRONG ANSWER',
+            Verdict.TIME_LIMIT_EXCEEDED: 'TIME LIMIT EXCEEDED',
+            Verdict.RUNTIME_ERROR: 'RUNTIME ERROR',
+            Verdict.COMPILER_ERROR: 'COMPILER ERROR',
+        }[self]
+
+
+def to_string(v: Verdict | None):
+    return {
+        Verdict.ACCEPTED: f'{Fore.GREEN}ACCEPTED{Style.RESET_ALL}',
+        Verdict.WRONG_ANSWER: f'{Fore.RED}WRONG ANSWER{Style.RESET_ALL}',
+        Verdict.TIME_LIMIT_EXCEEDED: f'{Fore.MAGENTA}TIME LIMIT EXCEEDED{Style.RESET_ALL}',
+        Verdict.RUNTIME_ERROR: f'{Fore.YELLOW}RUNTIME ERROR{Style.RESET_ALL}',
+        None: f'{Fore.BLUE}?{Style.RESET_ALL}',
+    }[v]
+
+
+def to_char(v: Verdict | None, lower=False):
+    return {
+        Verdict.ACCEPTED: f'{Fore.GREEN}A{Style.RESET_ALL}',
+        Verdict.WRONG_ANSWER: f'{Fore.RED}W{Style.RESET_ALL}',
+        Verdict.TIME_LIMIT_EXCEEDED: f'{Fore.MAGENTA}T{Style.RESET_ALL}',
+        Verdict.RUNTIME_ERROR: f'{Fore.YELLOW}R{Style.RESET_ALL}',
+        None: f'{Fore.BLUE}?{Style.RESET_ALL}',
+    }[v]
+
+
+def from_string(s: str) -> Verdict:
+    match s:
+        case 'CORRECT' | 'ACCEPTED' | 'AC':
+            return Verdict.ACCEPTED
+        case 'WRONG-ANSWER' | 'WRONG_ANSWER' | 'WA':
+            return Verdict.WRONG_ANSWER
+        case 'TIMELIMIT' | 'TIME_LIMIT_EXCEEDED' | 'TLE':
+            return Verdict.TIME_LIMIT_EXCEEDED
+        case 'RUN-ERROR' | 'RUN_TIME_ERROR' | 'RUNTIME_ERROR' | 'RTE':
```

Comment:
I appreciate your effort to fix this long-standing grammatical error, but DOMjudge only supports `RUN-ERROR` and `RUN_TIME_ERROR`: https://github.com/DOMjudge/domjudge/blob/main/webapp/src/Service/SubmissionService.php#L40

Similarly, the short-hands `AC`/`WA`/... will not be supported by DOMjudge (unless we send them a PR, of course).

Reply:
Oh, but `from_string` is only used to translate Various Other Naming Conventions to the internal `verdicts.Verdict`. Never in the other direction. The ambition is to _allow_ VANC to continue to exist.

Reply:
That's fair! In that case, I've updated `Submission._get_expected_verdicts` to enforce compatibility with DOMjudge :slightly_smiling_face: